### ISSUE TITLE
generated d.ts file located at wrong path

### ISSUE THREAD

**Issue by heimwege** [odata2ts/odata2ts#179]
Hi @texttechne,

sorry for finding again something strange 🙈 
My config looks as follows:

```
import { Modes, EmitModes } from "@odata2ts/odata2ts";
import type { ConfigFileOptions } from "@odata2ts/odata2ts";

const config: ConfigFileOptions = {
    debug: true,
    mode: Modes.models,
    emitMode: EmitModes.dts,
    skipEditableModels: false,
    skipIdModels: false,
    skipOperations: false,
    services: {
        TaskManagementService: {
            source: "webapp/localService/metadata.xml",
            output: "webapp/types/gen"
        }
    }
};

export default config;

```

and this is where the generated d.ts file gets stored 

![image](https://github.com/odata2ts/odata2ts/assets/73878785/d7126b11-535a-4a66-b4bb-d68adb53e16d)

Somehow parts of the given output path are duplicated. I would have expected the file in `webapp/types/gen/TaskManagementService.d.ts` but got `webapp/types/gen/types/gen/TaskManagementService.d.ts` and now I'm wondering if I have an error in my config or this is a bug. I also tried only with `webapp` and can confirm the wipe works perfectly 😹 

This is the log:
```
Loaded config file:  odata2ts.config.ts
---------------------------
Starting generation process
Resolved config: {
  source: 'webapp/localService/metadata.xml',
  output: 'webapp/types/gen',
  debug: true,
  mode: 'models',
  emitMode: 'dts',
  prettier: false,
  serviceName: undefined
}
Reading file: webapp/localService/metadata.xml
Emitting declaration files
```

Yet Another Test
`output: "webapp/foo"` results in `webapp/foo/foo/TaskManagementService.d.ts`

**Comment by texttechne** [comment]
This come up once. Can you please check if the explanation in [this issue](https://github.com/odata2ts/odata2ts/issues/137) is correct?

**Comment by heimwege** [comment]
I should have had a closer look at the closed issues. Sorry for that. I'll try if this helps. Looks promising 👍

**Comment by zinserjan** [comment]
@heimwege I had the same problem today. I generated an fiori elements application with the fiori tools generator in VSCode, tried to setup odata2ts and this issue appeared.

I found two solutions:

1. pretty simple and not recommended: If you want to emit the types within the `rootDir`, just remove the rootDir option in tsconfig.json. But I wouldn't recommended this, cause this allows you to import files outside of rootDir.
2. use a secondary tsconfig for odata2ts as @texttechne suggested (can be empty to apply only defaults)

**odata2ts.config.ts**
```
import { ConfigFileOptions, EmitModes, Modes } from "@odata2ts/odata2ts";

const config: ConfigFileOptions = {
    mode: Modes.models,
    emitMode: EmitModes.dts,
    skipEditableModels: true,
    tsconfig: "tsconfig.odata2ts.json",
    services: {
        orderManage: {
            source: "webapp/localService/metadata.xml",
            serviceName: "OrderManage",
            output: "webapp/types/odata"
        }
    }
};
export default config;

```
**tsconfig.odata2ts.json**
```
{}
```

**Comment by texttechne** [comment]
Ok, as it turns out, the mentioned issue is not related... sorry @heimwege for leading you on the wrong trail...

The culprit for the duplicated path is that `odata2ts` uses most options from your `tsconfig.json` and also sets the outputPath to the one configured in `odata2ts.config.ts`. And if the `tsconfig.json` specifies the `rootDir`, then duplicate paths occur:
- rootDir (set by you): copy files **including their paths** starting from this dir to the output folder
- outputDir (set by odata2ts): copy files to the configured folder

### Solution
Actually `odata2ts` must / should ignore the `rootDir` (& `rootDirs`) setting of your `tsconfig.json`.

So it's solution 1 that @zinserjan presents. And while it's less than ideal for you to drop the rootDir option, it's absolutely fine for odata2ts.

@zinserjan Thanks!!!

**Comment by texttechne** [comment]
Released with `0.25.1` of `@odata2ts/odata2ts`.

@heimwege please test and close the issue if it works.

**Comment by heimwege** [comment]
Works like a charm. Thanks a lot @texttechne  👍
